content/news/buzz/2025-curtin-capstone-semester-end-blog/contents.lr

```diff
@@ -0,0 +1,109 @@
+  Toga does not currently support testing for the Web platform, due to the fact that WebAssembly does not support multithreading. Currently, the test suite is a second thread inside the testbed application, which requires multithreading.This poses a few problems as the testbed is built assuming that these are running in the same process.
+
+  First, the probes that need to test the rendered widget in the web browser will be using a testing tool called Playwright. Playwright will allow us to ‘poke’ or ‘probe’ the DOM state of the app, looking at different attributes, styling, hierarchy, etc.
+
+  Second, there has to be some kind of communication bridge between the test suite and the app. We are currently looking at using Pyodide to run Python in the browser context remotely using Playwright. Though we will have to expose Pyodide when in test mode, as it is currently inaccessible when running the app normally.
+
+  Lastly, [about proxies]
+
+  …tbd…
+
+  ### PyScript Briefcase and Toga Dependencies
+
+  During Mitchell’s development of the Date Input Web Widget, we discovered some issues resulting from create_proxy. This issue was resolved separately, but led to discussions regarding ownership of Pyscript, as well as Shoelace and Bootstrap, within the wider Beeware architecture. Currently, the Briefcase Web Template inserts these into index.html, although these should be dependencies of Toga.
+
+  Russell has previously attempted this in the pull requests [https://github.com/beeware/briefcase/pull/1285](https://github.com/beeware/briefcase/pull/1285), [https://github.com/beeware/toga/pull/1945](https://github.com/beeware/toga/pull/1945) and [https://github.com/beeware/briefcase-web-static-template/pull/9](https://github.com/beeware/briefcase-web-static-template/pull/9). Currently, we plan on reviving parts of these to allow for Toga to specify Pyscript versioning. Discussion is ongoing regarding decoupling Pyscript from Briefcase and handling multiple dependencies.
+
+  More details can be found in our [Discussion Post on Briefcase.](https://github.com/beeware/briefcase/discussions/2283)
+
+  ### Briefcase Web Development Optimisations
+
+  While contributing to Toga’s web widgets, we noticed that testing even small changes requires rebuilding all project wheels, which can take a significant amount of time. This makes local development slow and interrupts the feedback loop needed for efficient frontend iteration.
+
+  To improve this, we’re exploring ways to optimise how Briefcase handles builds.
+
+  Ideas:
+  - A selective rebuild mechanism that only regenerates wheels when relevant files have changed
+  - A `--dev` implementation for web that skips wheel building and instead directly references source files during development.
+
+  We’ve outlined this in more detail in our [Discussion post on Briefcase](https://github.com/beeware/briefcase/discussions/2282) and are actively researching and working on prototypes
+
+---
+pub_date: 2025-06-XX
```

Comment:
Lets assume for now this will be published at the end of the semester; otherwise the page won't render at all.
```suggestion
pub_date: 2025-06-16
```